gammapy/stats/variability.py

```diff
@@ -82,3 +84,76 @@ def compute_chisq(flux):
     yobs = flux.data
     chi2, pval = stats.chisquare(yobs, yexp)
     return chi2, pval
+
+
+def compute_dtime(flux, flux_err, time, axis=0):
+    r"""Calculate the characteristic doubling time for a series of measurements.
+
+    The characteristic doubling time  is estimated to obtain the
+    minimum variability timescale for the light curves in which
+    rapid variations are clearly evident: for example it is useful in AGN flaring episodes.
+
+    This quantity, especially for AGN flares, is often expressed
+    as the pair of doubling time and halving time, or the minimum characteristic time
+    for the rising and falling components respectively.
+
+    Parameters
+    ----------
+    flux : `~astropy.units.Quantity`
+        the measured fluxes
+    flux_err : `~astropy.units.Quantity`
+        the error on measured fluxes
+    time: `~astropy.units.Quantity`
+        the times at which the fluxes are measured
+    axis : int, optional
+        Axis along which the value is computed.
+        The default is to compute the value on axis 0.
+
+    Returns
+    -------
+    dtime, dtime_err : `~numpy.ndarray`
+        Characteristic doubling time, halving time and errors.
+
+    References
+    ----------
+    ..[Brown2013] "Locating the γ-ray emission region
+    of the flat spectrum radio quasar PKS 1510−089", Brown et al. (2013)
+    https://academic.oup.com/mnras/article/431/1/824/1054498
+    """
+
+    flux = np.atleast_2d(flux).swapaxes(0, axis).T
+    flux_err = np.atleast_2d(flux_err).swapaxes(0, axis).T
+
+    times = (time[1:] - time[:-1]) / np.log2(flux[..., 1:] / flux[..., :-1])
```

Comment:
In this case, I suppose passing time edges is necessary, not time centers?

Reply:
My point was that from `compute_lightcurve_dtime` you pass the time centers. As per your logic here, you should be passing the edges, no?

Reply:
Sorry for the delay in the response. I don't think any between edges (min or max) or centers are perfect. I would still prefer to pass the centers of the bins in this case as they better represent an "average coordinate" of the flux point. Sorry if I am missing something obvious, but why are edges more logic here?